src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8202,6 +8239,17 @@ namespace WaterThermalTanks {
                 }
 
                 if (TfinalDiff < TemperatureConvergenceCriteria) break;
+
+                if (Tank.DesuperheaterNum > 0){
+                    DesuperheaterPLR = WaterHeaterDesuperheater(Tank.DesuperheaterNum).DesuperheaterPLR;
+                    DesuperheaterHeaterRate = WaterHeaterDesuperheater(Tank.DesuperheaterNum).HeaterRate;
+                    MdotDesuperheaterWater = WaterHeaterDesuperheater(Tank.DesuperheaterNum).OperatingWaterFlowRate * RhoH2O(Tavg[Tank.SourceOutletStratNode - 1]);
+                    if (DesuperheaterPLR > 0.0){
+                        Tank.SourceInletTemp = Tavg[Tank.SourceOutletStratNode - 1] + (DesuperheaterHeaterRate/DesuperheaterPLR) / (MdotDesuperheaterWater * Cp);
```

Comment:
Is there any chance that DesuperheaterPLR comes through with PLR > 0 but MdotDesuperheaterwater is zero causing a divide by zero?

Reply:
Added one condition check